### PR TITLE
litex: remove minimal build option

### PR DESCRIPTION
**Pull request by xobs**
This removes `CIRCUITPY_MINIMAL_BUILD ` and adds config options to get circuitpython building on Fomu.

It also adds required functions such as `mp_hal_delay_us` and the `os` module, which are necessary for doing a non-minimal build.

### CODE REVIEW

File: ports/litex/common-hal/os/__init__.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+#include "genhdr/mpversion.h"
+#include "py/mpconfig.h"
+#include "py/objstr.h"
+#include "py/objtuple.h"
+#include "py/qstr.h"
+
+#ifdef SAMD51
+#include "hal/include/hal_rand_sync.h"
+#endif
```
Comment (tannewt) [inline ports/litex/common-hal/os/__init__.c:35]: Remove please.

Comment (xobs) [reply ports/litex/common-hal/os/__init__.c]: Good catch.  Removed.

**Comment by tannewt** [comment]
Looks like you need to merge in master. ulab changed but github's merge doesn't look smart enough to update the submodule. Not sure why it only broke fomu though.

**Comment by xobs** [comment]
I did a rebase onto master.  Let's see if that fixes it.

**Comment by xobs** [comment]
The issue was the micropython config was missing `MICROPY_PY_REVERSE_SPECIAL_METHODS`.  This has now been added.

**Pull request merged**